app/src/main/java/net/skyscanner/backpack/demo/compose/CarouselStory.kt

```diff
@@ -0,0 +1,112 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.demo.compose
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.aspectRatio
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import com.google.accompanist.pager.ExperimentalPagerApi
+import net.skyscanner.backpack.compose.carousel.BpkCarousel
+import net.skyscanner.backpack.compose.carousel.rememberBpkCarouselState
+import net.skyscanner.backpack.compose.text.BpkText
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+import net.skyscanner.backpack.demo.R
+
+@Composable
+@Preview
+fun CarouselStory(
+  modifier: Modifier = Modifier,
+) {
+  val currentIndex by remember { mutableStateOf(1) }
+
+  Column(
+    modifier = modifier
+      .fillMaxSize()
+      .padding(horizontal = BpkSpacing.Base, vertical = BpkSpacing.Base)
+      .verticalScroll(rememberScrollState()),
+    horizontalAlignment = Alignment.CenterHorizontally,
+  ) {
+
+    BpkText(text = stringResource(id = R.string.carousel_only_1_item))
+    CarouselSample(totalImages = 1)
+
+    BpkText(text = stringResource(id = R.string.carousel_multiple_items))
+    CarouselSample(totalImages = 3)
+
+    BpkText(text = stringResource(id = R.string.carousel_start_at_different_index))
+    CarouselSample(totalImages = 3, initialImage = 2)
+
+    BpkText(text = stringResource(id = R.string.carousel_image_change_callback))
+    CarouselSample(totalImages = 3, initialImage = currentIndex, showCurrentPageLabel = true)
+  }
+}
+
+@OptIn(ExperimentalPagerApi::class)
+@Composable
+private fun CarouselSample(
+  totalImages: Int,
+  modifier: Modifier = Modifier,
+  initialImage: Int = 0,
+  showCurrentPageLabel: Boolean = false,
+) {
+  val pagerState = rememberBpkCarouselState(totalImages = totalImages, initialImage = initialImage)
+
+  if (showCurrentPageLabel) {
+    BpkText(
+      text = stringResource(id = R.string.carousel_current_image).format(pagerState.currentPage),
```

Comment:
You can just use stringResource(R.string.carousel_current_image, pagerState.currentPage)